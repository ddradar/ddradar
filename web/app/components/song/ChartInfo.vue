<script setup lang="ts">
import type { StepChart } from '@ddradar/core'
import { difficultyMap, playStyleMap } from '@ddradar/core'

import type { AccordionItem } from '#ui/types'

const props = defineProps<{ chart: StepChart }>()

const { t } = useI18n()
const title = computed(() => {
  const playStyleText = playStyleMap.get(props.chart.playStyle)!
  const difficultyName = difficultyMap.get(props.chart.difficulty)!
  const shortPlayStyle = `${playStyleText[0]}P` as 'SP' | 'DP'
  return `${shortPlayStyle}-${difficultyName} (${props.chart.level})` as const
})
const color = computed(() => getChartColor(props.chart.difficulty))
const items = computed<AccordionItem[]>(() => [
  { label: title.value, slot: 'chart' },
  { label: t('score'), slot: 'score' },
])
</script>

<template>
  <UAccordion :color="color" :items="items">
    <template #chart>
      <div class="grid grid-cols-2 h-300">
        <div class="m-2">
          <ul>
            <li><em>Notes</em>: {{ chart.notes }}</li>
            <li><em>Freeze Arrow</em>: {{ chart.freezeArrow }}</li>
            <li><em>Shock Arrow</em>: {{ chart.shockArrow }}</li>
            <br />
          </ul>
        </div>
      </div>
    </template>
  </UAccordion>
</template>

<i18n lang="json">
{
  "ja": {
    "score": "スコアランキング",
    "chart": "譜面情報"
  },
  "en": {
    "score": "Score Board",
    "chart": "Chart Info"
  }
}
</i18n>
