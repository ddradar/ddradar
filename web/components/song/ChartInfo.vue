<script setup lang="ts">
import type { CourseChartSchema, StepChartSchema } from '@ddradar/core'
import { difficultyMap, playStyleMap } from '@ddradar/core'

import type { AccordionItem } from '#ui/types'

const props = defineProps<{ chart: CourseChartSchema | StepChartSchema }>()

const { t } = useI18n()
const title = computed(() => {
  /* eslint-disable @typescript-eslint/no-non-null-assertion */
  const playStyleText = playStyleMap.get(props.chart.playStyle)!
  const difficultyName = difficultyMap.get(props.chart.difficulty)!
  /* eslint-enable @typescript-eslint/no-non-null-assertion */
  const shortPlayStyle = `${playStyleText[0]}P` as 'SP' | 'DP'
  return `${shortPlayStyle}-${difficultyName} (${props.chart.level})` as const
})
const color = computed(() => getChartColor(props.chart.difficulty))
const items = computed<AccordionItem[]>(() => [
  { label: t('score'), slot: 'score' },
  { label: t('chart'), slot: 'chart' },
  ...(isCourseChart(props.chart)
    ? [{ label: t('course'), slot: 'course' }]
    : []),
])
</script>

<template>
  <UCard
    :ui="{ body: { base: '', padding: 'px-0 py-2 sm:px-0 sm:py-2' } }"
    :color="color"
  >
    <template #header>
      {{ title }}
    </template>
    <UAccordion :color="color" :items="items">
      <template #chart>
        <ul>
          <li><em>Notes</em>: {{ chart.notes }}</li>
          <li><em>Freeze Arrow</em>: {{ chart.freezeArrow }}</li>
          <li><em>Shock Arrow</em>: {{ chart.shockArrow }}</li>
        </ul>
      </template>
    </UAccordion>
  </UCard>
</template>

<i18n lang="json">
{
  "ja": {
    "score": "スコアランキング",
    "chart": "譜面情報",
    "course": "コース曲目"
  },
  "en": {
    "score": "Score Board",
    "chart": "Chart Info",
    "course": "Course Order"
  }
}
</i18n>
