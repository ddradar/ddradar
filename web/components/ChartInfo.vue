<template>
  <section
    class="column is-half-tablet is-one-third-desktop is-one-quarter-widescreen"
  >
    <Card title="Chart Info" :variant="cardType" collapsible>
      <div class="card-content">
        <div class="content">
          <ul>
            <li><em>Notes</em>: {{ chart.notes }}</li>
            <li><em>Freeze Arrow</em>: {{ chart.freezeArrow }}</li>
            <li><em>Shock Arrow</em>: {{ chart.shockArrow }}</li>
          </ul>
          <ol v-if="isCourse(chart)">
            <li v-for="o in chart.order" :key="o.songId">
              <nuxt-link
                :to="`/songs/${o.songId}#${o.playStyle}-${o.difficulty}`"
              >
                {{ o.songName }}
              </nuxt-link>
              {{ getChartTitle(o) }}
            </li>
          </ol>
        </div>
        <div v-if="!isCourse(chart)" class="table-container">
          <table class="table">
            <thead>
              <tr>
                <th><abbr title="STREAM">STR</abbr></th>
                <th><abbr title="VOLTAGE">VOL</abbr></th>
                <th>AIR</th>
                <th><abbr title="FREEZE">FRE</abbr></th>
                <th><abbr title="CHAOS">CHA</abbr></th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>{{ chart.stream }}</td>
                <td>{{ chart.voltage }}</td>
                <td>{{ chart.air }}</td>
                <td>{{ chart.freeze }}</td>
                <td>{{ chart.chaos }}</td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </Card>
  </section>
</template>

<script lang="ts" setup>
import { Song } from '@ddradar/core'
import { computed } from 'vue'

import Card from '~/components/Card.vue'
import type { CourseInfo } from '~/server/api/v1/courses/[id].get'
import type { SongInfo } from '~/server/api/v1/songs/[id].get'
import { difficultyMap, getChartTitle } from '~/src/song'

type CourseChart = CourseInfo['charts'][number]
type Chart = SongInfo['charts'][number] | CourseChart

interface ChartInfoProps {
  chart: Chart
}

const props = defineProps<ChartInfoProps>()

const cardType = computed(
  () =>
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    difficultyMap
      .get(props.chart.difficulty)!
      .toLowerCase() as Lowercase<Song.DifficultyName>
)

const isCourse = (chart: Chart): chart is CourseChart =>
  Array.isArray((chart as CourseChart).order)
</script>
