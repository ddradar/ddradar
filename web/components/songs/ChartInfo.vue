<template>
  <section
    class="column is-half-tablet is-one-third-desktop is-one-quarter-widescreen"
  >
    <ScoreBoard :song-id="songId" :is-course="isCourse" :chart="chart" />
    <CollapsibleCard title="Chart Info" :variant="cardType" collapsible>
      <div class="card-content">
        <div class="content">
          <ul>
            <li><em>Notes</em>: {{ chart.notes }}</li>
            <li><em>Freeze Arrow</em>: {{ chart.freezeArrow }}</li>
            <li><em>Shock Arrow</em>: {{ chart.shockArrow }}</li>
          </ul>
          <ol v-if="isCourse">
            <li
              v-for="o in (chart as CourseInfo['charts'][number]).order"
              :key="o.songId"
            >
              <nuxt-link
                :to="`/songs/${o.songId}#${o.playStyle}-${o.difficulty}`"
              >
                {{ o.songName }}
              </nuxt-link>
              {{ getChartTitle(o) }}
            </li>
          </ol>
        </div>
        <div v-if="!isCourse" class="table-container">
          <table class="table">
            <thead>
              <tr>
                <th><abbr title="STREAM">STR</abbr></th>
                <th><abbr title="VOLTAGE">VOL</abbr></th>
                <th>AIR</th>
                <th><abbr title="FREEZE">FRE</abbr></th>
                <th><abbr title="CHAOS">CHA</abbr></th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>{{ (chart as SongInfo['charts'][number]).stream }}</td>
                <td>{{ (chart as SongInfo['charts'][number]).voltage }}</td>
                <td>{{ (chart as SongInfo['charts'][number]).air }}</td>
                <td>{{ (chart as SongInfo['charts'][number]).freeze }}</td>
                <td>{{ (chart as SongInfo['charts'][number]).chaos }}</td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </CollapsibleCard>
  </section>
</template>

<script lang="ts" setup>
import { difficultyMap } from '@ddradar/core'

import CollapsibleCard from '~~/components/CollapsibleCard.vue'
import ScoreBoard from '~~/components/songs/ScoreBoard.vue'
import type { CourseInfo } from '~~/server/api/v1/courses/[id].get'
import type { SongInfo } from '~~/server/api/v1/songs/[id].get'
import { getChartTitle } from '~~/utils/song'

type Chart = SongInfo['charts'][number] | CourseInfo['charts'][number]

interface ChartInfoProps {
  songId: SongInfo['id']
  chart: Chart
}

const props = defineProps<ChartInfoProps>()

const cardType = computed(() =>
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  difficultyMap.get(props.chart.difficulty)!.toLowerCase()
)
const isCourse = computed(() =>
  Array.isArray((props.chart as CourseInfo['charts'][number]).order)
)
</script>
