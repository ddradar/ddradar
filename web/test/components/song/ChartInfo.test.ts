import { mountSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, test } from 'vitest'
import { createI18n } from 'vue-i18n'

import { testCourseData, testSongData } from '~/../core/test/data'
import ChartInfo from '~/components/song/ChartInfo.vue'
import { locales } from '~/test/test-utils'

describe('components/song/ChartInfo.vue', () => {
  describe.each(locales)('{ locale: "%s" }', locale => {
    const global = { plugins: [createI18n({ locale, legacy: false })] }

    test.each([...testSongData.charts, ...testCourseData.charts])(
      '{ chart: %o } snapshot test',
      async chart => {
        // Arrange - Act
        const wrapper = await mountSuspended(ChartInfo, {
          global,
          props: { chart },
        })
        // Assert
        expect(wrapper.element).toMatchSnapshot()
      }
    )
  })
})
