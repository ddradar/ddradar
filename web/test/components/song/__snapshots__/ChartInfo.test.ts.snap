// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/song/ChartInfo.vue > { locale: "en" } > { chart: { playStyle: 1, difficulty: +0, level: 4, notes: 138, freezeArrow: +0, shockArrow: +0, stream: 29, voltage: 22, air: 5, freeze: +0, chaos: +0 } } snapshot test 1`] = `
<div
  class="w-full flex flex-col"
>
  
  <div
    class="w-full flex flex-col"
    data-headlessui-state=""
  >
    <button
      aria-controls="headlessui-disclosure-panel-nsvOHSTtmF4_1"
      aria-expanded="false"
      class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 text-blue-500 dark:text-blue-400 bg-blue-50 hover:bg-blue-100 disabled:bg-blue-50 dark:bg-blue-950 dark:hover:bg-blue-900 dark:disabled:bg-blue-950 focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-blue-500 dark:focus-visible:ring-blue-400 inline-flex items-center mb-1.5 w-full"
      data-headlessui-state=""
      id="headlessui-disclosure-button-nsvOHSTtmF4_0"
      type="button"
    >
      
      
      <!--v-if-->
      
      
      <span
        class=""
      >
        SP-BEGINNER (4)
      </span>
      
      
      <svg
        aria-hidden="true"
        class="icon h-5 w-5 ms-auto transform transition-transform duration-200"
        data-v-cf6c0647=""
        height="1em"
        role="img"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <path
          clip-rule="evenodd"
          d="M5.22 8.22a.75.75 0 0 1 1.06 0L10 11.94l3.72-3.72a.75.75 0 1 1 1.06 1.06l-4.25 4.25a.75.75 0 0 1-1.06 0L5.22 9.28a.75.75 0 0 1 0-1.06"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      
      
    </button>
    <transition-stub
      appear="false"
      css="true"
      enteractiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      leaveactiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      persisted="false"
    >
      <div
        style="display: none;"
      >
        <div
          class="text-sm text-gray-500 dark:text-gray-400 pt-1.5 pb-3"
          data-headlessui-state=""
          id="headlessui-disclosure-panel-nsvOHSTtmF4_1"
        >
          
          <div
            class="grid grid-cols-2 h-300"
          >
            <div
              class="m-2"
            >
              <ul>
                <li>
                  <em>
                    Notes
                  </em>
                  : 138
                </li>
                <li>
                  <em>
                    Freeze Arrow
                  </em>
                  : 0
                </li>
                <li>
                  <em>
                    Shock Arrow
                  </em>
                  : 0
                </li>
                <br />
              </ul>
            </div>
          </div>
          
        </div>
      </div>
    </transition-stub>
  </div>
  <div
    class="w-full flex flex-col"
    data-headlessui-state=""
  >
    <button
      aria-controls="headlessui-disclosure-panel-nsvOHSTtmF4_3"
      aria-expanded="false"
      class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 text-blue-500 dark:text-blue-400 bg-blue-50 hover:bg-blue-100 disabled:bg-blue-50 dark:bg-blue-950 dark:hover:bg-blue-900 dark:disabled:bg-blue-950 focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-blue-500 dark:focus-visible:ring-blue-400 inline-flex items-center mb-1.5 w-full"
      data-headlessui-state=""
      id="headlessui-disclosure-button-nsvOHSTtmF4_2"
      type="button"
    >
      
      
      <!--v-if-->
      
      
      <span
        class=""
      >
        Score Board
      </span>
      
      
      <svg
        aria-hidden="true"
        class="icon h-5 w-5 ms-auto transform transition-transform duration-200"
        data-v-cf6c0647=""
        height="1em"
        role="img"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <path
          clip-rule="evenodd"
          d="M5.22 8.22a.75.75 0 0 1 1.06 0L10 11.94l3.72-3.72a.75.75 0 1 1 1.06 1.06l-4.25 4.25a.75.75 0 0 1-1.06 0L5.22 9.28a.75.75 0 0 1 0-1.06"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      
      
    </button>
    <transition-stub
      appear="false"
      css="true"
      enteractiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      leaveactiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      persisted="false"
    >
      <div
        style="display: none;"
      >
        <div
          class="text-sm text-gray-500 dark:text-gray-400 pt-1.5 pb-3"
          data-headlessui-state=""
          id="headlessui-disclosure-panel-nsvOHSTtmF4_3"
        >
          
          
          
        </div>
      </div>
    </transition-stub>
  </div>
  
</div>
`;

exports[`components/song/ChartInfo.vue > { locale: "en" } > { chart: { playStyle: 1, difficulty: 1, level: 8, notes: 264, freezeArrow: +0, shockArrow: +0, stream: 56, voltage: 44, air: 18, freeze: +0, chaos: 4 } } snapshot test 1`] = `
<div
  class="w-full flex flex-col"
>
  
  <div
    class="w-full flex flex-col"
    data-headlessui-state=""
  >
    <button
      aria-controls="headlessui-disclosure-panel-nsvOHSTtmF4_5"
      aria-expanded="false"
      class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 text-yellow-500 dark:text-yellow-400 bg-yellow-50 hover:bg-yellow-100 disabled:bg-yellow-50 dark:bg-yellow-950 dark:hover:bg-yellow-900 dark:disabled:bg-yellow-950 focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-yellow-500 dark:focus-visible:ring-yellow-400 inline-flex items-center mb-1.5 w-full"
      data-headlessui-state=""
      id="headlessui-disclosure-button-nsvOHSTtmF4_4"
      type="button"
    >
      
      
      <!--v-if-->
      
      
      <span
        class=""
      >
        SP-BASIC (8)
      </span>
      
      
      <svg
        aria-hidden="true"
        class="icon h-5 w-5 ms-auto transform transition-transform duration-200"
        data-v-cf6c0647=""
        height="1em"
        role="img"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <path
          clip-rule="evenodd"
          d="M5.22 8.22a.75.75 0 0 1 1.06 0L10 11.94l3.72-3.72a.75.75 0 1 1 1.06 1.06l-4.25 4.25a.75.75 0 0 1-1.06 0L5.22 9.28a.75.75 0 0 1 0-1.06"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      
      
    </button>
    <transition-stub
      appear="false"
      css="true"
      enteractiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      leaveactiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      persisted="false"
    >
      <div
        style="display: none;"
      >
        <div
          class="text-sm text-gray-500 dark:text-gray-400 pt-1.5 pb-3"
          data-headlessui-state=""
          id="headlessui-disclosure-panel-nsvOHSTtmF4_5"
        >
          
          <div
            class="grid grid-cols-2 h-300"
          >
            <div
              class="m-2"
            >
              <ul>
                <li>
                  <em>
                    Notes
                  </em>
                  : 264
                </li>
                <li>
                  <em>
                    Freeze Arrow
                  </em>
                  : 0
                </li>
                <li>
                  <em>
                    Shock Arrow
                  </em>
                  : 0
                </li>
                <br />
              </ul>
            </div>
          </div>
          
        </div>
      </div>
    </transition-stub>
  </div>
  <div
    class="w-full flex flex-col"
    data-headlessui-state=""
  >
    <button
      aria-controls="headlessui-disclosure-panel-nsvOHSTtmF4_7"
      aria-expanded="false"
      class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 text-yellow-500 dark:text-yellow-400 bg-yellow-50 hover:bg-yellow-100 disabled:bg-yellow-50 dark:bg-yellow-950 dark:hover:bg-yellow-900 dark:disabled:bg-yellow-950 focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-yellow-500 dark:focus-visible:ring-yellow-400 inline-flex items-center mb-1.5 w-full"
      data-headlessui-state=""
      id="headlessui-disclosure-button-nsvOHSTtmF4_6"
      type="button"
    >
      
      
      <!--v-if-->
      
      
      <span
        class=""
      >
        Score Board
      </span>
      
      
      <svg
        aria-hidden="true"
        class="icon h-5 w-5 ms-auto transform transition-transform duration-200"
        data-v-cf6c0647=""
        height="1em"
        role="img"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <path
          clip-rule="evenodd"
          d="M5.22 8.22a.75.75 0 0 1 1.06 0L10 11.94l3.72-3.72a.75.75 0 1 1 1.06 1.06l-4.25 4.25a.75.75 0 0 1-1.06 0L5.22 9.28a.75.75 0 0 1 0-1.06"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      
      
    </button>
    <transition-stub
      appear="false"
      css="true"
      enteractiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      leaveactiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      persisted="false"
    >
      <div
        style="display: none;"
      >
        <div
          class="text-sm text-gray-500 dark:text-gray-400 pt-1.5 pb-3"
          data-headlessui-state=""
          id="headlessui-disclosure-panel-nsvOHSTtmF4_7"
        >
          
          
          
        </div>
      </div>
    </transition-stub>
  </div>
  
</div>
`;

exports[`components/song/ChartInfo.vue > { locale: "ja" } > { chart: { playStyle: 1, difficulty: +0, level: 4, notes: 138, freezeArrow: +0, shockArrow: +0, stream: 29, voltage: 22, air: 5, freeze: +0, chaos: +0 } } snapshot test 1`] = `
<div
  class="w-full flex flex-col"
>
  
  <div
    class="w-full flex flex-col"
    data-headlessui-state=""
  >
    <button
      aria-controls="headlessui-disclosure-panel-nsvOHSTtmF4_9"
      aria-expanded="false"
      class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 text-blue-500 dark:text-blue-400 bg-blue-50 hover:bg-blue-100 disabled:bg-blue-50 dark:bg-blue-950 dark:hover:bg-blue-900 dark:disabled:bg-blue-950 focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-blue-500 dark:focus-visible:ring-blue-400 inline-flex items-center mb-1.5 w-full"
      data-headlessui-state=""
      id="headlessui-disclosure-button-nsvOHSTtmF4_8"
      type="button"
    >
      
      
      <!--v-if-->
      
      
      <span
        class=""
      >
        SP-BEGINNER (4)
      </span>
      
      
      <svg
        aria-hidden="true"
        class="icon h-5 w-5 ms-auto transform transition-transform duration-200"
        data-v-cf6c0647=""
        height="1em"
        role="img"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <path
          clip-rule="evenodd"
          d="M5.22 8.22a.75.75 0 0 1 1.06 0L10 11.94l3.72-3.72a.75.75 0 1 1 1.06 1.06l-4.25 4.25a.75.75 0 0 1-1.06 0L5.22 9.28a.75.75 0 0 1 0-1.06"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      
      
    </button>
    <transition-stub
      appear="false"
      css="true"
      enteractiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      leaveactiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      persisted="false"
    >
      <div
        style="display: none;"
      >
        <div
          class="text-sm text-gray-500 dark:text-gray-400 pt-1.5 pb-3"
          data-headlessui-state=""
          id="headlessui-disclosure-panel-nsvOHSTtmF4_9"
        >
          
          <div
            class="grid grid-cols-2 h-300"
          >
            <div
              class="m-2"
            >
              <ul>
                <li>
                  <em>
                    Notes
                  </em>
                  : 138
                </li>
                <li>
                  <em>
                    Freeze Arrow
                  </em>
                  : 0
                </li>
                <li>
                  <em>
                    Shock Arrow
                  </em>
                  : 0
                </li>
                <br />
              </ul>
            </div>
          </div>
          
        </div>
      </div>
    </transition-stub>
  </div>
  <div
    class="w-full flex flex-col"
    data-headlessui-state=""
  >
    <button
      aria-controls="headlessui-disclosure-panel-nsvOHSTtmF4_11"
      aria-expanded="false"
      class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 text-blue-500 dark:text-blue-400 bg-blue-50 hover:bg-blue-100 disabled:bg-blue-50 dark:bg-blue-950 dark:hover:bg-blue-900 dark:disabled:bg-blue-950 focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-blue-500 dark:focus-visible:ring-blue-400 inline-flex items-center mb-1.5 w-full"
      data-headlessui-state=""
      id="headlessui-disclosure-button-nsvOHSTtmF4_10"
      type="button"
    >
      
      
      <!--v-if-->
      
      
      <span
        class=""
      >
        スコアランキング
      </span>
      
      
      <svg
        aria-hidden="true"
        class="icon h-5 w-5 ms-auto transform transition-transform duration-200"
        data-v-cf6c0647=""
        height="1em"
        role="img"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <path
          clip-rule="evenodd"
          d="M5.22 8.22a.75.75 0 0 1 1.06 0L10 11.94l3.72-3.72a.75.75 0 1 1 1.06 1.06l-4.25 4.25a.75.75 0 0 1-1.06 0L5.22 9.28a.75.75 0 0 1 0-1.06"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      
      
    </button>
    <transition-stub
      appear="false"
      css="true"
      enteractiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      leaveactiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      persisted="false"
    >
      <div
        style="display: none;"
      >
        <div
          class="text-sm text-gray-500 dark:text-gray-400 pt-1.5 pb-3"
          data-headlessui-state=""
          id="headlessui-disclosure-panel-nsvOHSTtmF4_11"
        >
          
          
          
        </div>
      </div>
    </transition-stub>
  </div>
  
</div>
`;

exports[`components/song/ChartInfo.vue > { locale: "ja" } > { chart: { playStyle: 1, difficulty: 1, level: 8, notes: 264, freezeArrow: +0, shockArrow: +0, stream: 56, voltage: 44, air: 18, freeze: +0, chaos: 4 } } snapshot test 1`] = `
<div
  class="w-full flex flex-col"
>
  
  <div
    class="w-full flex flex-col"
    data-headlessui-state=""
  >
    <button
      aria-controls="headlessui-disclosure-panel-nsvOHSTtmF4_13"
      aria-expanded="false"
      class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 text-yellow-500 dark:text-yellow-400 bg-yellow-50 hover:bg-yellow-100 disabled:bg-yellow-50 dark:bg-yellow-950 dark:hover:bg-yellow-900 dark:disabled:bg-yellow-950 focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-yellow-500 dark:focus-visible:ring-yellow-400 inline-flex items-center mb-1.5 w-full"
      data-headlessui-state=""
      id="headlessui-disclosure-button-nsvOHSTtmF4_12"
      type="button"
    >
      
      
      <!--v-if-->
      
      
      <span
        class=""
      >
        SP-BASIC (8)
      </span>
      
      
      <svg
        aria-hidden="true"
        class="icon h-5 w-5 ms-auto transform transition-transform duration-200"
        data-v-cf6c0647=""
        height="1em"
        role="img"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <path
          clip-rule="evenodd"
          d="M5.22 8.22a.75.75 0 0 1 1.06 0L10 11.94l3.72-3.72a.75.75 0 1 1 1.06 1.06l-4.25 4.25a.75.75 0 0 1-1.06 0L5.22 9.28a.75.75 0 0 1 0-1.06"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      
      
    </button>
    <transition-stub
      appear="false"
      css="true"
      enteractiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      leaveactiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      persisted="false"
    >
      <div
        style="display: none;"
      >
        <div
          class="text-sm text-gray-500 dark:text-gray-400 pt-1.5 pb-3"
          data-headlessui-state=""
          id="headlessui-disclosure-panel-nsvOHSTtmF4_13"
        >
          
          <div
            class="grid grid-cols-2 h-300"
          >
            <div
              class="m-2"
            >
              <ul>
                <li>
                  <em>
                    Notes
                  </em>
                  : 264
                </li>
                <li>
                  <em>
                    Freeze Arrow
                  </em>
                  : 0
                </li>
                <li>
                  <em>
                    Shock Arrow
                  </em>
                  : 0
                </li>
                <br />
              </ul>
            </div>
          </div>
          
        </div>
      </div>
    </transition-stub>
  </div>
  <div
    class="w-full flex flex-col"
    data-headlessui-state=""
  >
    <button
      aria-controls="headlessui-disclosure-panel-nsvOHSTtmF4_15"
      aria-expanded="false"
      class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 text-yellow-500 dark:text-yellow-400 bg-yellow-50 hover:bg-yellow-100 disabled:bg-yellow-50 dark:bg-yellow-950 dark:hover:bg-yellow-900 dark:disabled:bg-yellow-950 focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-yellow-500 dark:focus-visible:ring-yellow-400 inline-flex items-center mb-1.5 w-full"
      data-headlessui-state=""
      id="headlessui-disclosure-button-nsvOHSTtmF4_14"
      type="button"
    >
      
      
      <!--v-if-->
      
      
      <span
        class=""
      >
        スコアランキング
      </span>
      
      
      <svg
        aria-hidden="true"
        class="icon h-5 w-5 ms-auto transform transition-transform duration-200"
        data-v-cf6c0647=""
        height="1em"
        role="img"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <path
          clip-rule="evenodd"
          d="M5.22 8.22a.75.75 0 0 1 1.06 0L10 11.94l3.72-3.72a.75.75 0 1 1 1.06 1.06l-4.25 4.25a.75.75 0 0 1-1.06 0L5.22 9.28a.75.75 0 0 1 0-1.06"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      
      
    </button>
    <transition-stub
      appear="false"
      css="true"
      enteractiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      leaveactiveclass="overflow-hidden transition-[height] duration-200 ease-out"
      persisted="false"
    >
      <div
        style="display: none;"
      >
        <div
          class="text-sm text-gray-500 dark:text-gray-400 pt-1.5 pb-3"
          data-headlessui-state=""
          id="headlessui-disclosure-panel-nsvOHSTtmF4_15"
        >
          
          
          
        </div>
      </div>
    </transition-stub>
  </div>
  
</div>
`;
